### ISSUE TITLE
SSR Sveltekit Flashing ( Solved )

### ISSUE THREAD

**Issue by impactvelocity** [svelteuidev/svelteui#172]
### What package has an issue

@svelteuidev/core

### A clear and concise description of what the bug is

This is a stitches SSR issue, I really hope this is the solution, it seems to work.

ISSUE

The css will flash or pop in after the page is loaded. This would happen on Cloudflare Pages, Vercel edge and serverless run times, and it was driving me insane.

I made a gif of what would happen: https://share.getcloudapp.com/6quz4RAQ 

Sometimes the CSS would load fine, no flash, or a super quick unnoticable, I open it up in a bunch of tabs, and the last time you can see it jump. Some of the css is fine, and some loads in later. Very annoying.

I printed the getCssText on the page, and you can see it does change: https://share.getcloudapp.com/2NumZkox 

I am hitting refresh, you can see the css starts with what looks like the theme then loads all the other styles.

SOLUTION

For whatever reason, it seems the getCssText() on SSR doesn't seem to be 100% correct until full client side page load.

Idea was to hooks.js to inject the css inside of the HTML which seemed to work! 

`src/app.html` I added `%stitches-css-ssr%` in the head placeholder 

then inside `src/hooks.js `I swapped that out with get the css from stitches.

```
import { getCssText } from '@svelteuidev/core'

export async function handle({ event, resolve }) {
	const response = await resolve(event, {
		transformPageChunk: ({ html }) => {
			return html.replace('%stitches-css-ssr%', `<style id="stitches">${getCssText()}</style>`)
		}
	})

	return response
}

```

That seems to get the latest CssText from stitches correctly and injects it. I haven't seen a flash or weird
load in yet. Tried on CF Pages, and Vercel Edge + Serverless in two projects, it seems to work!

( took me way too long to figure this out )

### In which browser(s) did the problem occur?

Chrome

### Steps To Reproduce

Try with SSR

<SvelteUIProvider withNormalizeCSS withGlobalStyles ssr>
 

And see it it flashes in 

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by Brisklemonade** [comment]
Thank you for exploring one of the SSR issues. We initially thought that injecting it through a component would be enough. We soon after found out that's not the case, so this would take a bit more time to solve. Using `hooks` is a good idea, and I think making a helper function to just drop in a project would be the way to go with that.

If you would like to work on this, we would love the help! We can either use this issue thread, or discuss this on discord if you would like.

**Comment by drewbitt** [comment]
Sveltekit has hit 1.0 so I hope some focus can be spent here (also mentioned in #107). The library is still FOUCing with the default Sveltekit rendering which probably makes it unusable for many applications. Is this fixed in the "Upgrade to latest SvelteKit version (leverage Vite 3.0.0)" completed item for 0.8 @Brisklemonade ?

**Comment by BeeMargarida** [comment]
One other possible solution can be to add the line below to the root `+layout.svelte`. No mater the solution, this needs to be added to the svelteUI documentation

```
<svelte:head>
	{@html `<style id="stitches">${getCssText()}</style>`}
</svelte:head>
```

**Comment by BeeMargarida** [comment]
Should be fixed in the new minor version 0.8.0
